ui/src/network.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2021 The Radicle Upstream Contributors
+//
+// This file is part of radicle-upstream, distributed under the GPLv3
+// with Radicle Linking Exception. For full terms see the included
+// LICENSE file.
+
+import type { SvelteComponent } from "svelte";
+import { Status, StatusType } from "ui/src/localPeer";
+
+import Icon from "ui/DesignSystem/Icon";
+import SyncingIcon from "ui/DesignSystem/Sidebar/ConnectionStatusIndicator/Syncing.svelte";
+import OfflineIcon from "ui/DesignSystem/Sidebar/ConnectionStatusIndicator/Offline.svelte";
+
+const connectedPeerCount = (peers: { [peerId: string]: string[] }): string => {
+  const count = Object.keys(peers).length;
+  return peerCount(count);
+};
+
+const peerCount = (count: number): string => {
+  if (count === 1) {
+    return "1 peer";
+  } else {
+    return `${count} peers`;
+  }
+};
+
+interface IndicatorState {
+  text: string;
+  cy: string;
+  icon: typeof SvelteComponent;
+  fill: string;
+}
+
+export const indicatorState = (status: Status): IndicatorState | undefined => {
```

Comment:
We want to omit the `undefined` case from the return type. It should never happen because you covered all status types. Unfortunately Typescript is not smart enough (yet) to figure this out on its own so we need to add the following `else` statement at the end
```typescript
} else {
  return unreachable(status)
}
```
with `import { unreachable } from `ui/src/unreachable`. Then you can remove `undefined` from the return type.

Reply:
https://github.com/radicle-dev/radicle-upstream/pull/2066/commits/841cb0998cff274d0adfc34bb2b6b6bb458a8df9